SpatialGDK/Source/SpatialGDK/Private/Interop/ActorSystem.cpp

```diff
@@ -0,0 +1,1833 @@
+
+				FSpatialNetDeltaSerializeInfo::DeltaSerializeRead(NetDriver, ValueDataReader, ReplicatedObject, Parent->ArrayIndex,
+																  Parent->Property, NetDeltaStruct);
+
+				ObjectReferences.MappedRefs.Append(NewMappedRefs);
+			}
+			else
+			{
+				TSet<FUnrealObjectRef> NewMappedRefs;
+				TSet<FUnrealObjectRef> NewUnresolvedRefs;
+				FSpatialNetBitReader BitReader(NetDriver->PackageMap, ObjectReferences.Buffer.GetData(), ObjectReferences.NumBufferBits,
+											   NewMappedRefs, NewUnresolvedRefs);
+				check(Property->IsA<GDK_PROPERTY(StructProperty)>());
+
+				bool bHasUnresolved = false;
+				ReadStructProperty(BitReader, GDK_CASTFIELD<GDK_PROPERTY(StructProperty)>(Property), NetDriver, Data + AbsOffset,
+								   bHasUnresolved);
+
+				ObjectReferences.MappedRefs.Append(NewMappedRefs);
+			}
+
+			if (Parent && Parent->Property->HasAnyPropertyFlags(CPF_RepNotify))
+			{
+				if (Parent->RepNotifyCondition == REPNOTIFY_Always || !Property->Identical(StoredData + StoredDataOffset, Data + AbsOffset))
+				{
+					RepNotifies.AddUnique(Parent->Property);
+				}
+			}
+		}
+	}
+}
+
+void ActorSystem::OnHeartbeatComponentUpdate(const Worker_EntityId EntityId, Schema_ComponentUpdate* Update)
+{
+	if (!NetDriver->IsServer())
+	{
+		// Clients can ignore Heartbeat component updates.
+		return;
+	}
+
+	TWeakObjectPtr<USpatialNetConnection>* ConnectionPtr = AuthorityPlayerControllerConnectionMap.Find(EntityId);
+	if (ConnectionPtr == nullptr)
+	{
+		// Heartbeat component update on a PlayerController that this server does not have authority over.
+		// TODO: Disable component interest for Heartbeat components this server doesn't care about - UNR-986
+		return;
+	}
+
+	if (!ConnectionPtr->IsValid())
+	{
+		UE_LOG(LogActorSystem, Warning,
+			   TEXT("Received heartbeat component update after NetConnection has been cleaned up. PlayerController entity: %lld"),
+			   EntityId);
+		AuthorityPlayerControllerConnectionMap.Remove(EntityId);
+		return;
+	}
+
+	USpatialNetConnection* NetConnection = ConnectionPtr->Get();
+
+	Schema_Object* EventsObject = Schema_GetComponentUpdateEvents(Update);
+	uint32 EventCount = Schema_GetObjectCount(EventsObject, SpatialConstants::HEARTBEAT_EVENT_ID);
+	if (EventCount > 0)
+	{
+		if (EventCount > 1)
+		{
+			UE_LOG(LogActorSystem, Verbose, TEXT("Received multiple heartbeat events in a single component update, entity %lld."),
+				   EntityId);
+		}
+
+		NetConnection->OnHeartbeat();
+	}
+
+	Schema_Object* FieldsObject = Schema_GetComponentUpdateFields(Update);
+	if (Schema_GetBoolCount(FieldsObject, SpatialConstants::HEARTBEAT_CLIENT_HAS_QUIT_ID) > 0
+		&& GetBoolFromSchema(FieldsObject, SpatialConstants::HEARTBEAT_CLIENT_HAS_QUIT_ID))
+	{
+		// Client has disconnected, let's clean up their connection.
+		CloseClientConnection(NetConnection, EntityId);
+	}
+}
+
+USpatialActorChannel* ActorSystem::GetOrRecreateChannelForDormantActor(AActor* Actor, const Worker_EntityId EntityID) const
```

Comment:
I prefer my incorrect spelling

Reply:
if it wouldn't have my name next to in the blame, I'd leave it